raiden/network/rpc/client.py

```diff
@@ -244,46 +229,18 @@ def block_number(self):
         """ Return the most recent block. """
         return self.web3.eth.blockNumber
 
-    def nonce_needs_update(self):
-        if self.nonce_available_value is None:
-            return True
-
-        now = time.time()
-
-        # Python's 2.7 time is not monotonic and it's affected by clock resets,
-        # force an update.
-        if self.nonce_last_update > now:
-            return True
-
-        return now - self.nonce_last_update > self.nonce_update_interval
-
-    def nonce_update_from_node(self):
-        nonce = -2
-        nonce_available_value = self.nonce_available_value or -1
-
-        # Wait until all tx are registered as pending
-        while nonce < nonce_available_value:
-            pending_transactions = self.web3.eth.getTransactionCount(
-                to_checksum_address(self.sender),
-            )
-            nonce = pending_transactions + self.nonce_offset
-
-            log.debug(
-                'updated nonce from server',
-                server=nonce,
-                local=nonce_available_value,
-            )
-
-        self.nonce_last_update = time.time()
-        self.nonce_available_value = nonce
+    @cachedmethod(attrgetter('_nonce_cache'))
+    def _node_nonce_it(self) -> Iterable[int]:
+        pending_transactions = self.web3.eth.getTransactionCount(
```

Comment:
`pending_transactions` is the wrong name for this. Just use `transaction_count`. It is essentially the number of already completed transactions for the account.

Reply:
Well, not only completed, but pending too (therefore the name of the variables). This `getTransactionCount` needs to include both the number of mined transactions sent by this account plus the tx sent to this node's pool, required to keep the `nonce` synced even when multiple transactions are sent in a single block. Just to mention here, but this may enable the parallel channel creation on join I tried to implement and debug long time ago, which may decrease token network join times by as much as 10x. May work on that as an optimization sometime in the future, as it's quite simple to implement.

Reply:
Still `pending_transactions` give the wrong idea. The RPC call used by all DAPPS and defined in the [JSON RPC Spec](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_gettransactioncount) gives the number of the transactions sent from this account.

Please just use `transaction_count` as it encompasses both meanings (both pending and already completed).